config/kafka.go

```diff
@@ -0,0 +1,120 @@
+package config
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/asyncapi/event-gateway/asyncapi"
+	v2 "github.com/asyncapi/event-gateway/asyncapi/v2"
+	"github.com/asyncapi/event-gateway/kafka"
+	"github.com/pkg/errors"
+)
+
+// KafkaProxy holds the config for later configuring a Kafka proxy.
+type KafkaProxy struct {
+	BrokerFromServer   string              `split_words:"true"`
+	BrokersMapping     pipeSeparatedValues `split_words:"true"`
+	BrokersDialMapping pipeSeparatedValues `split_words:"true"`
+	ExtraFlags         pipeSeparatedValues `split_words:"true"`
+}
+
+// ProxyConfig creates a config struct for the Kafka Proxy based on a given AsyncAPI doc (if provided).
+func (c *KafkaProxy) ProxyConfig(doc []byte, debug bool) (kafka.ProxyConfig, error) {
+	var kafkaProxyConfig kafka.ProxyConfig
+	if len(doc) == 0 && len(c.BrokersMapping.Values) == 0 {
+		return kafkaProxyConfig, errors.New("either AsyncAPIDoc or KafkaProxyBrokersMapping config should be provided")
+	}
+
+	if c.BrokerFromServer != "" && len(doc) == 0 {
+		return kafkaProxyConfig, errors.New("AsyncAPIDoc should be provided when setting BrokerFromServer")
+	}
+
+	if len(doc) > 0 {
+		conf, err := c.configFromDoc(doc)
+		if err != nil {
+			return kafkaProxyConfig, err
+		}
+
+		kafkaProxyConfig = conf
+	} else {
+		kafkaProxyConfig = kafka.ProxyConfig{
+			BrokersMapping:     c.BrokersMapping.Values,
+			DialAddressMapping: c.BrokersDialMapping.Values,
+		}
+	}
+
+	kafkaProxyConfig.Debug = debug
+	kafkaProxyConfig.ExtraConfig = c.ExtraFlags.Values
+
+	if err := kafkaProxyConfig.Validate(); err != nil {
+		return kafkaProxyConfig, err
+	}
+
+	return kafkaProxyConfig, nil
+}
+
+func (c *KafkaProxy) configFromDoc(d []byte) (kafka.ProxyConfig, error) {
+	var kafkaProxyConfig kafka.ProxyConfig
+
+	doc := new(v2.Document)
+	if err := v2.Decode(d, doc); err != nil {
+		return kafkaProxyConfig, errors.Wrap(err, "error decoding AsyncAPI json doc to Document struct")
+	}
+
+	var err error
+	if c.BrokerFromServer != "" {
+		kafkaProxyConfig, err = kafkaProxyConfigFromServer(c.BrokerFromServer, doc)
+	} else {
+		kafkaProxyConfig, err = kafkaProxyConfigFromAllServers(doc)
+	}
+
+	return kafkaProxyConfig, err
+}
+
+func isValidKafkaProtocol(s asyncapi.Server) bool {
+	return strings.HasPrefix(s.Protocol(), "kafka")
+}
```

Comment:
Maybe this should be in `kafka/utils.go` file?

Reply:
Based on your suggestion, I moved all related to config to `config.go` including that function, used **only** for validating the config.

Reply:
You didn't move that function 😄  I mean, at the moment we have `config/kafka.go` and `kafka/config.go` where in first place you create the config, but in second you validate it. I understand that it's probably a problem with a private fields for `config` module, but yeah, it's a bit weird :)

Reply:
Oh sorry my fault. I don't know why I read this comment was mentioning the `kafka` package instead.

Regarding your suggestion, note that the package `kafka` is completely agnostic from any AsyncAPI particularity. `kafka` proxy only knows about server mappings, etc. 

Regarding the creation/validation process, the purpose of postponing validation is to centralize it in just a single point. Doing the validation in a constructor, for example, won't be so nice as there are several parameters, most of them optional, so calling the constructor with tons of `nil` is not that ok. 
As I still want to centralize validation (of the proxy, not talking about the whole app), I think it's best to implement it with functional options:
Functional options is nowadays a common pattern in Go: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

With this, I'm able to just require only mandatory params in the constructor and leave the rest of the trivial config to those options.
Look at my latest commit. Now validation is happening when calling `kafka.NewProxyConfig`.

Reply:
Ok, now it's better, thanks! :)